solana-programs/audius_eth_registry/src/processor.rs

```diff
@@ -27,14 +36,127 @@ impl Processor {
     /// ValidSigner version indicating signer uninitialization
     pub const VALID_SIGNER_UNINITIALIZED_VERSION: u8 = 0;
 
+    /// Process [recover instruction data]().
+    pub fn recover_instruction_data(
+        signature_data: &SignatureData,
+        valid_signer: &ValidSigner,
+    ) -> Vec<u8> {
```

Comment:
Check out the new updated Claimable Tokens signature validation. Actually big chunks of the current code are not required. Check out this much more compact signature verification code: https://github.com/atticwip/claimable-tokens/blob/41680cbf3b0efe1bfe0f5c32af78892602132c1a/src/processor.rs#L123

Reply:
it's not wrong to compare the signatures, this limits the code to only the valid segments

Reply:
https://github.com/AudiusProject/audius-protocol/pull/1530/files